fix: re-enable peer ID inlining but make it configurable

Stebalien opened the pull request:
This should:

1. Unblock OpenBazaar. They were relying on the non-inlining behavior but run a forked network so they can set `EnableDeprecatedKeyInlining` to `false`.
2. Unblock Textile. They rely on this feature being *enabled*. Re-enabling it by default will give us some time to transition away from it.

Stebalien commented:
@hoffmabc does this work for OpenBazaar? See: https://github.com/libp2p/specs/issues/138

Stebalien commented:
cc @cpacia

Stebalien commented:
CC @parkan.

parkan commented:
also paging @placer14

placer14 commented:
I wasn't aware of the conversation. I'll raise it with the team and get back to you guys. Thanks for the consideration. 🥇

Stebalien commented:
I've changed the flag to `AdvancedEnableInlining` and added a bunch of warnings/caveats to make it clear that this isn't something users should touch unless they know what they're doing.

Stebalien commented:
Merging as this simply restores the current behavior.

sanderpick commented:
Nice, cheers @Stebalien

Stebalien commented:
@sanderpick this has been merged into master, along with the DHT dialing fixes.

sanderpick commented:
Okay, great. I'm under full siege this week... will update our node with go-ipfs master next week.
